Add ability to specify name in file extension client

pmm-sumo opened the pull request:
**Description:** 

This is a proposal how to make File Extension more generic. When used e.g. for WAL purposes this will allow to have separate client for each signal type

pmm-sumo committed:
Add ability to specify name in file extension client

pmm-sumo commented:
@djaglowski this is a quick draft that adds `name` as another attribute to `GetClient`, giving a generic ability to use separate storage instances for the same type of component. What do you think?

djaglowski commented on extension/storage/README.md:
````diff
@@ -4,7 +4,7 @@ A storage extension persists state beyond the collector process. Other component
 The `storage.Extension` interface extends `component.Extension` by adding the following method:
 ```
-GetClient(component.Kind, component.Kind, config.ComponentID) (Client, error)
+GetClient(component.Kind, component.Kind, config.ComponentID, string) (Client, error)
````
Looks like there was a preexisting typo here.
```suggestion
GetClient(context.Context, component.Kind, config.ComponentID, string) (Client, error)
```

djaglowski reviewed (approved):
Looks good to me.

pmm-sumo committed:
Address typo in README and include name in tests

pmm-sumo commented:
I have added a simple test-case for single component having two clients and addressed the README.md fix. Build seems to fail on an unrelated load-test at the moment
